### PR TITLE
Add note vision subsystem

### PR DESCRIPTION
**Pull request by mimizh2418**

### CODE REVIEW

File: src/main/java/org/team1540/robot2024/subsystems/vision/gamepiece/NoteVisionIO.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.littletonrobotics.junction.AutoLog;
+
+public interface NoteVisionIO {
+    @AutoLog
+    class NoteVisionIOInputs {
+        public boolean hasDetection = false;
+        public double lastDetectionTimestampSecs = 0.0;
+        public double targetPitchRads = 0.0;
+        public double targetYawRads = 0.0;
+        public double targetAreaRads = 0.0;
```
Comment (WeilSimon) [inline src/main/java/org/team1540/robot2024/subsystems/vision/gamepiece/NoteVisionIO.java:12]: I find it hard to believe that this is in radians

File: src/main/java/org/team1540/robot2024/subsystems/vision/gamepiece/NoteVisionIO.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package org.team1540.robot2024.subsystems.vision.gamepiece;
+
+import org.littletonrobotics.junction.AutoLog;
+
+public interface NoteVisionIO {
+    @AutoLog
+    class NoteVisionIOInputs {
+        public boolean hasDetection = false;
+        public double lastDetectionTimestampSecs = 0.0;
+        public double targetPitchRads = 0.0;
```
Comment (WeilSimon) [inline src/main/java/org/team1540/robot2024/subsystems/vision/gamepiece/NoteVisionIO.java:10]: Why not make these rotation 2ds?

File: src/main/java/org/team1540/robot2024/subsystems/vision/gamepiece/NoteVisionIO.java (hunk 1 above)
Comment (mimizh2418) [reply src/main/java/org/team1540/robot2024/subsystems/vision/gamepiece/NoteVisionIO.java:12]: :skull:

File: src/main/java/org/team1540/robot2024/Constants.java [hunk 3]
```diff
@@ -134,6 +132,7 @@ public static class AprilTag {
         public static class Gamepiece {
             public static final String CAMERA_NAME = "limelight-notevision";
```
Comment (WeilSimon) [inline src/main/java/org/team1540/robot2024/Constants.java:133]: Make name shorter

**Review by WeilSimon — CHANGES_REQUESTED** [review]

**Pull request merged**